src/Razor/src/Microsoft.AspNetCore.Razor.VSCode/src/BlazorDebugConfigurationProvider.ts

```diff
@@ -0,0 +1,64 @@
+/* --------------------------------------------------------------------------------------------
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See License.txt in the project root for license information.
+ * ------------------------------------------------------------------------------------------ */
+
+import * as vscode from 'vscode';
+
+import { RazorLogger } from './RazorLogger';
+
+export class BlazorDebugConfigurationProvider implements vscode.DebugConfigurationProvider {
+    private logger: RazorLogger;
+    private vscodeType: typeof vscode;
+
+    constructor(logger: RazorLogger, vscodeType: typeof vscode) {
+        this.logger = logger;
+        this.vscodeType = vscodeType;
+    }
+
+    public resolveDebugConfiguration(folder: vscode.WorkspaceFolder | undefined, configuration: vscode.DebugConfiguration): vscode.ProviderResult<vscode.DebugConfiguration> {
+        const app = {
+            name: '.NET Core Launch (Blazor Standalone)',
+            type: 'coreclr',

```

Comment:
Is there any advantage to spawning this program under the debugger instead of [child_process.spawn](https://nodejs.org/api/child_process.html#child_process_child_process_spawn_command_args_options)?

Reply:
To provide parity with the existing experience and to allow users to view output from their restore/run steps within VS Code. Are there downsides to this that would outweigh those benefits?

Reply:
The main reason I would advise against it is this just feels like a very heavy weight way of getting log messages -- seems like we should just use `child_process` and log those messages to some VS Code output pain rather than running the debugger, which is quite a large and sophisticated piece of code and has no idea that you don't care about all that sophistication.

The other reason I don't like this way of running the web server is there is no way for users to pass along any other options to vsdbg, including the options that the [debugger team depends on to get logs](https://github.com/OmniSharp/omnisharp-vscode/wiki/Enabling-C%23-debugger-logging). So if something goes wrong with it we will have no way of supporting users.

Reply:
I see. Thanks for that write up. I'll take a look at changing this to launch the dev server process instead of using the debug methods.